File: packages/shared/src/types/domain.ts
/**
 * Domain types that mirror the Python API models exactly.
 * Source of truth: apps/api/app/models/domain.py
 */

export type TitleLevel =
  | "c_suite"
  | "founder"
  | "vp"
  | "director"
  | "manager"
  | "individual"
  | "unknown";

export type SourceType =
  | "wikidata"
  | "wikipedia"
  | "company_site"
  | "sec_filing"
  | "university_page";

export interface Institution {
  id: string;
  name: string;
  slug: string;
  aliases: string[];
  country: string | null;
  wikidata_id: string | null;
}

export interface Company {
  id: string;
  name: string;
  slug: string;
  sector: string | null;
  domain: string | null;
}

export interface Employment {
  company: Company | null;
  title: string;
  title_level: TitleLevel;
  sector: string;
  is_current: boolean;
}

export interface Person {
  id: string;
  full_name: string;
  source_url: string;
  source_type: SourceType;
  retrieved_at: string;
  confidence: number;
  verified_fields: string[];
}

export interface SearchResult {
  person: Person;
  employment: Employment[];
  institution: Institution | null;
}

export interface SearchResponse {
  results: SearchResult[];
  total: number;
  page: number;
  limit: number;
  institution: Institution | null;
}

export interface UniversityResponse {
  institution: Institution;
  alumni_count: number;
  top_employers: Array<{ company: string; count: number }>;
  sector_breakdown: Record<string, number>;
  title_level_breakdown: Record<string, number>;
}

export interface SearchFilters {
  university: string;
  sector?: string;
  title_level?: TitleLevel;
  company_type?: string;
  region?: string;
  keyword?: string;
  page?: number;
  limit?: number;
}

File: apps/api/app/models/domain.py
"""Core domain types. Pydantic models for safe IO."""
from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel, Field

TitleLevel = Literal["c_suite", "founder", "vp", "director", "manager", "individual", "unknown"]
SourceType = Literal["wikidata", "wikipedia", "company_site", "sec_filing", "university_page"]


class Institution(BaseModel):
    id: str
    name: str
    slug: str
    aliases: list[str] = Field(default_factory=list)
    country: Optional[str] = None
    wikidata_id: Optional[str] = None


class Company(BaseModel):
    id: str
    name: str
    slug: str
    sector: Optional[str] = None
    domain: Optional[str] = None


class Employment(BaseModel):
    company: Optional[Company] = None
    title: str = ""
    title_level: TitleLevel = "unknown"
    sector: str = "other"
    is_current: bool = False


class Person(BaseModel):
    id: str
    full_name: str
    source_url: str
    source_type: SourceType
    retrieved_at: datetime
    confidence: float = 0.5
    verified_fields: list[str] = Field(default_factory=list)


class SearchResult(BaseModel):
    person: Person
    employment: list[Employment] = Field(default_factory=list)
    institution: Optional[Institution] = None
